Sources/ArgumentParser/Parsable Properties/CompletionKind.swift

```diff
@@ -9,56 +9,225 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// The type of completion to use for an argument or option.
+/// The type of completion to use for an argument or option value.
+///
+/// For all `CompletionKind`s, the completion shell script is configured with
+/// the following settings, which will not affect the requesting shell outside
+/// the completion script:
+///
+/// - bash:
+///
+///   ```shell
+///   shopt -s extglob
+///   set +o history +o posix
+///   ```
+///
+/// - fish: no settings
+///
+/// - zsh:
+///
+///   ```shell
+///   emulate -RL zsh -G
+///   setopt extendedglob
+///   unsetopt aliases banghist
+///   ```
 public struct CompletionKind {
+  /// The type of completion to use for an argument or option value.
```

Comment:
We can skip the doc comments on the enum and its cases, since they're all internal.